modules/rest_api/controllers/services/rest.php

```diff
@@ -1131,14 +1155,36 @@ private function getEsPostData($postObj, $format, $file) {
         elseif ($field === '#data_cleaner_icons#') {
           $fields[] = 'identification.auto_checks';
         }
-        elseif ($field === '#datasource_code#') {
+        elseif (preg_match('/^#datasource_code(:.+)*#$/', $field)) {
           $fields[] = 'metadata.website';
           $fields[] = 'metadata.survey';
+          $fields[] = 'metadata.group';
         }
         elseif ($field === '#event_date#') {
           $fields[] = 'event.date_start';
           $fields[] = 'event.date_end';
         }
+        elseif (preg_match('/^#record_date(.*)#$/', $field)) {
```

Comment:
This may be a separate change to make in future, but I suspect a single preg_match to capture the parts of a special field expression could be used, then the series of if statements can just test the special field name in a variable, rather than re-doing lots of similar preg_matches.

Reply:
I'll take a look.